test/LEGO.AsyncAPI.Tests/Bindings/Sns/SnsBindings_Should.cs

```diff
@@ -77,22 +86,38 @@ public void SnsChannelBinding_WithFilledObject_SerializesAndDeserializes()
                         new Statement()
                         {
                             Effect = Effect.Deny,
-                            Principal = new StringOrStringList(new AsyncApiAny("arn:aws:iam::123456789012:user/alex.wichmann")),
+                            Principal = new Principal(new PrincipalStar()),
```

Comment:
one thing that stands out to me, is this api.
Where we create a principal with another principal.

Im guessing this is purely due to the parsing logic, but perhaps this could be made more succint.
So the api becomes

```
Principal = new PrincipalStar()
```
the same for the principal object.